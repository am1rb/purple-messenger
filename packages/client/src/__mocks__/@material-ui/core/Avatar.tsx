import React from "react";
import type { AvatarProps } from "@material-ui/core/Avatar";

function MuiAvatar({ src, alt }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} alt={alt} />;
}

export default MuiAvatar;
