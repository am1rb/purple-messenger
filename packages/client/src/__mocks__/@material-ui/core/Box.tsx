import React from "react";

export interface BoxProps {
  children: React.ReactNode;
}

const muiBox = jest.fn(({ children }: BoxProps) => (
  <div data-testid="mui-box">{children}</div>
));

export default muiBox;
