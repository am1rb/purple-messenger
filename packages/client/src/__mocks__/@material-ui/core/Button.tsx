import React from "react";

export interface ButtonProps {
  onClick?: () => void;
  children: React.ReactNode;
}

function MuiButton({ onClick, children }: ButtonProps) {
  return (
    <div data-testid="mock-mui-button" onClick={onClick}>
      {children}
    </div>
  );
}

export default MuiButton;
