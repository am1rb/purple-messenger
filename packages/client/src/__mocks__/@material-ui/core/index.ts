export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
