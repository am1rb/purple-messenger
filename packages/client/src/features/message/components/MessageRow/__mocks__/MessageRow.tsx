import React from "react";

const messageRow = jest.fn(() => <div data-testid="mock-message-row" />);

export default messageRow;
