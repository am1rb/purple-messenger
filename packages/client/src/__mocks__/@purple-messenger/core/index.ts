import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
