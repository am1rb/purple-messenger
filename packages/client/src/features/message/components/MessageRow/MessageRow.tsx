import React, { memo } from "react";
import { Box, Card, Typography } from "@material-ui/core";
import {
  Message,
  MessageOwner,
  MessageStatus as MessageStatusEnum,
} from "@purple-messenger/core";
import MessageStatus from "../MessageStatus";
import MessageSentAt from "../MessageSentAt";
import { MessageSeenSensor } from "../MessageSeenSensor";

export type Props = Message;

function MessageRow({ body, status, owner, sentAt, id }: Props) {
  const message = (
    <Box
      m={2}
      maxWidth="65%"
      marginLeft={owner === MessageOwner.Me ? "auto" : undefined}
      marginRight={owner === MessageOwner.Friend ? "auto" : undefined}
    >
      <Card>
        <Box m={1}>
          <Typography gutterBottom variant="body2">
            {body}
          </Typography>
          <Box display="flex" justifyContent="space-between">
            {owner === MessageOwner.Me ? (
              <MessageStatus status={status} />
            ) : (
              <span />
            )}
            <MessageSentAt sentAt={sentAt} />
          </Box>
        </Box>
      </Card>
    </Box>
  );

  return owner === MessageOwner.Friend ? (
    <MessageSeenSensor status={status} messageId={id}>
      {message}
    </MessageSeenSensor>
  ) : (
    message
  );
}

export default memo(MessageRow);
