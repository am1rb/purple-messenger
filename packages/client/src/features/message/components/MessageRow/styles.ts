import { makeStyles } from "@material-ui/core";

export default makeStyles({ body: { whiteSpace: 'pre' } });