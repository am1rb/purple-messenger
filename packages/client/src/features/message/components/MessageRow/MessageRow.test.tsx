import React from "react";
import { render } from "@testing-library/react";
import { MessageOwner, MessageStatus } from "@purple-messenger/core";
import MessageRow, { MessageProps } from "./MessageRow";

jest.mock("../MessageStatus");
jest.mock("../MessageSentAt");
jest.mock("../MessageSeenSensor");

const sharedProps: MessageProps = {
  id: 1,
  body: "1021 Vucwas Place",
  owner: MessageOwner.Friend,
  sentAt: new Date(),
  status: MessageStatus.Sent,
  receivedAt: new Date(),
};

describe("The <MessageRow /> tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(<MessageRow {...sharedProps} />);
    expect(container).toMatchSnapshot();
  });

  it("Should render <MessageStatus /> if the owner is me", () => {
    const { getByTestId } = render(
      <MessageRow {...sharedProps} owner={MessageOwner.Me} />
    );
    expect(getByTestId("mock-message-status")).toBeInTheDocument();
  });

  it("Should not render <MessageStatus /> if the owner is friend", () => {
    const { queryByTestId } = render(
      <MessageRow {...sharedProps} owner={MessageOwner.Friend} />
    );
    expect(queryByTestId("mock-message-status")).not.toBeInTheDocument();
  });

  it("Should not render <MessageSeenSensor /> if the owner is me", () => {
    const { queryByTestId } = render(
      <MessageRow {...sharedProps} owner={MessageOwner.Me} />
    );
    expect(queryByTestId("mock-message-seen-sensor")).not.toBeInTheDocument();
  });

  it("Should render <MessageSeenSensor /> if the owner is friend", () => {
    const { getByTestId } = render(
      <MessageRow {...sharedProps} owner={MessageOwner.Friend} />
    );
    expect(getByTestId("mock-message-seen-sensor")).toBeInTheDocument();
  });
});
