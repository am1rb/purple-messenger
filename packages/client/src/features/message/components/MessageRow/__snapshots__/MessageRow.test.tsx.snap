// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <MessageRow /> tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mock-message-seen-sensor"
  />
</div>
`;
